api/v1beta1/common_types.go

```diff
@@ -160,7 +160,7 @@ const (
 	// only during a server side dry run apply operation. It is used for validating
 	// update webhooks for objects which get updated by template rotation (e.g. InfrastructureMachineTemplate).
 	// When the annotation is set and the admission request is a dry run, the webhook should
-	// deny validation due to immutability. By that the request will succeed (without
+	// skip validation due to immutability. By that the request will succeed (without
 	// any changes to the actual object because it is a dry run) and the topology controller
 	// will receive the resulting object.
```

Comment:
Maybe rephrase it even more, so it's clearer?

> TopologyDryRunAnnotation is an annotation that gets set on objects by the topology controller when performing a dry-run server side apply in order to apply defaulting to resources. It instructs the validating web hook to skip validation of immutable fields, which allows the request to succeed. The annotation must not be respected for regular (not dry-run) requests, so no invalid modifications to the resources are applied.
The topology controller uses this feature when processing immutable resources (usually templates) to decide whether a new revision of a immutable resource needs to be created.